### DISCUSSION TITLE
Using onChange removes focus as we type a cell

### DISCUSSION THREAD

**Discussion by rmrbytes** [iddan/react-spreadsheet#271] [Q&A]
Hello:

I am trying to use this library to edit some data (using Next.js). I am facing the following errors:

If I use the controlled method of input viz. using `<Spreadsheet data={data} onChange={setData} />` I find that everytime I enter a value in a cell, I lose control of the focus viz. I need to press enter key again to enter the cell to edit. 

Is this the expected behaviour?

Alternatively, if I do not use the onChange prop and wish to retrieve the entire matrix (assume using a button) I am unable to figure out how to use the API to get the spreadsheet data.

I will appreciate any help that can point me in the right direction.

Thanks

**Comment by iddan** [comment]
Hey, can you provide a small reproduction of this in CodeSandbox?

**Comment by rmrbytes** [comment]
@iddan: 

I realised that the onChange was causing a change in the props value to the component and that resulted in the spreadsheet component to get re-rendered.

My bad. I should have updated this discussion.

**Reply by federicoverly** [reply]
Heey! Did you manage to resolve the issue of constant re-rendering?
